folly/io/IOStreamBuf.h

```diff
@@ -0,0 +1,67 @@
+#include <memory>
+#include <streambuf>
+
+#include <folly/io/IOBuf.h>
+
+namespace folly {
+
+template<typename CharT, typename Traits = std::char_traits<CharT> >
+class IOStreamBuf : public std::basic_streambuf<CharT, Traits> {
+  // Due to having to merge single-byte subsets of CharT across IOBuf
+  // boundaries, prevent the use of IOStreamBuf on multi-byte types for now.
+  static_assert(sizeof(CharT) == 1,
+          "IOStreamBuf doesn't yet work with multi-byte types");
+
+ public:
+  /**
+   * Construct IOStreamBuf using the provided IOBuf, which may be chained.
+   * The IOStreamBuf does not own the IOBuf nor extend the lifetime of it; you
+   * must ensure that the IOBuf provided lasts at least as long as the
+   * IOStreamBuf.
+   */
+  IOStreamBuf(const folly::IOBuf* head);
```

Comment:
Via our internal linter:
> Single-argument constructor `IOStreamBuf(const folly::IOBuf* head)` may inadvertently be used as a type conversion constructor. Prefix the function with the `explicit` keyword to avoid this, or add an `/* implicit */` comment to suppress this warning.

In this case, marking it as `explicit` is the correct solution.